### PR TITLE
spidercoordinator: add new podCIDRType "auto"

### PR DESCRIPTION
**Pull request by cyclinder**
Add new podCIDRType "auto", It automatically switches podCIDRType based on the default CNI in the cluster. In addition, it improves the ability to get pod CIDR when default cni is cilium.

---
name: Pull request
about: Tell us about your contribution
labels: ["pr/release/none-required"]
---

---
## Thanks for contributing !

Before submitting a pull request, make sure you read about our Contribution notice here: <https://spidernet-io.github.io/spiderpool/contributing/development/pullrequest/#need-review>

**What this PR does / why we need it**:

Add new podCIDRType "auto", It automatically switches podCIDRType based on the default CNI in the cluster. In addition, it improves the ability to get pod CIDR when default cni is cilium.

**Which issue(s) this PR fixes**:

https://github.com/spidernet-io/spiderpool/issues/2287
https://github.com/spidernet-io/spiderpool/issues/2188

**Special notes for your reviewer**:

**make sure your commit is signed off**

### CODE REVIEW

**Comment by weizhoublue** [comment]
（1）需要有对应的 doc 说明 
每种 type 是什么意思，适应场景是什么
（2）当前对于检测出来的  cidr，是否有用例闭环

**Comment by cyclinder** [comment]
1. 文档已补
2. 现在e2e已经切换 auto, 不同CNI组合可以正常跑过，算是冒烟测试。具体 e2e case @ty-dc 在补

**Comment by weizhoublue** [comment]
> （1）需要有对应的 doc 说明
> 每种 type 是什么意思，适应场景是什么
> （2）当前对于检测出来的 cidr，是否有用例闭环

在 https://github.com/spidernet-io/spiderpool/blob/main/docs/usage/coordinator-zh_CN.md 中，没谈到 overlay 模式下，pod cidr 涉及到的 路由 同步问题

File: docs/usage/coordinator-zh_CN.md [hunk 1]
```diff
@@ -120,9 +120,11 @@ default via 10.6.0.1 dev eth0
 
 上述问题我们可以通过设置 coordinator 运行在 overlay 模式解决。 在此模式下，`coordinator` 不会创建 veth 设备，而是设置一些策略路由，确保 Pod 访问 ClusterIP 时从 eth0(通常由 Calico、Cilium等CNI创建) 转发，Pod 访问集群外部目标时从 net1 (通常由 Macvlan、IPvlan 等CNI创建) 转发。
 
+> 在 Overlay 模式下，Spiderpool 会自动同步集群缺省 CNI 的 Pod 子网，这些子网用于在多网卡 Pod 中设置路由，以实现它访问由缺省 CNI 创建的 Pod 之间的正常通信时，从 `eth0` 转发。这个配置对应 `spidercoordinator.spec.podCIDRType`，默认为 `auto`, 可选值: ["auto","calico","cilium","cluster","none"]。更多详情参考 [CRD-Spidercoordinator](../reference/crd-spidercoordinator.md)
+
```
Comment (weizhoublue) [inline docs/usage/coordinator-zh_CN.md:124]: 这些路由是在pod 启动时注入的，如果 相关的 cidr 发生了变动，是不能自动生效到 已经running 的pod中，这需要重启pod

Comment (cyclinder) [reply docs/usage/coordinator-zh_CN.md:128]: Done

**Pull request merged**